lib/smart_proxy_ansible/plugin.rb

```diff
@@ -1,16 +1,64 @@
+# frozen_string_literal: true
+
 module Proxy
   module Ansible
     # Calls for the smart-proxy API to register the plugin
     class Plugin < Proxy::Plugin
+
+      def runtime_dir
+        Pathname(ENV['RUNTIME_DIRECTORY'] || '/run').join(plugin_name.to_s)
+      end
+
+      def state_dir
+        Pathname(ENV['STATE_DIRECTORY'] || '/var/lib/foreman-proxy').join(plugin_name.to_s)
+      end
+
+      def cache_dir
+        Pathname(ENV['CACHE_DIRECTORY'] || '/var/cache/foreman-proxy').join(plugin_name.to_s)
+      end
+
+      def logs_dir
+        Pathname(ENV['LOGS_DIRECTORY'] || '/var/logs/foreman-proxy').join(plugin_name.to_s)
+      end
+
+      def config_dir
+        Pathname(ENV['CONFIGURATION_DIRECTORY'] || '/etc/foreman-proxy').join(plugin_name.to_s)
+      end
+
       rackup_path File.expand_path('http_config.ru', __dir__)
       settings_file 'ansible.yml'
       plugin :ansible, Proxy::Ansible::VERSION
       default_settings :ansible_dir => Dir.home,
-                       :ansible_environment_file => '/etc/foreman-proxy/ansible.env'
+                       :ansible_environment_file => '/etc/foreman-proxy/ansible.env',
+                       :vcs_integration => true,
+                       :static_roles_paths => %w[/etc/ansible/roles /usr/share/ansible/roles]
                        # :working_dir => nil
 
+      load_programmable_settings do |settings|
+        mutable_roles_path = state_dir.join('roles')
```

Comment:
I'm getting the following error:
`Disabling all modules in the group ['ansible'] due to a failure in one of them: undefined local variable or method 'state_dir' for Proxy::Ansible::Plugin:Class`

Reply:
I somehow missed the commit that turned the required methods into static ones. Should be fixed now.